File: src/pattern/template/java/StringDisplay.java
package pattern.template.java;

public class StringDisplay extends AbstractDisplay {
    String str;

    public StringDisplay(String str){
        this.str = str;
    }

    private void printLine(){
        System.out.println("+------------------+");
    }

    @Override
    public void open() {
        this.printLine();
    }

    @Override
    public void print() {
        System.out.println("+"+this.str+"+");
    }

    @Override
    public void close() {
        this.printLine();
    }
}

File: src/pattern/template/kotlin/StringDisplay.kt
package pattern.template.kotlin

class StringDisplay(var str: String): AbstractDisplay(){
    private fun printLine() {
        println("+-------------+")
    }

    override fun open() {
        printLine()
    }

    override fun print(){
        println("+${this.str}+")
    }

    override fun close() {
        printLine()
    }
}